### DISCUSSION TITLE
Rspamd protocol Hostname

### DISCUSSION THREAD

**Discussion by andy-igoshin** [rspamd/rspamd#5242] [General]
What is expected in the Hostname header ?

**Comment by moisseev** [comment]
The `Hostname` header in the Rspamd protocol represents the hostname of the client connecting to the mail transfer agent (MTA). This hostname is resolved by the MTA and then passed to Rspamd as part of the request.

**Comment by andy-igoshin** [comment]
You mean reverse dns resolving of client ip address?

**Reply by moisseev** [reply]
Yes, that's correct. However, the behavior may depend on the MTA configuration.

**Comment by andy-igoshin** [comment]
Thanks!

**Comment by andy-igoshin** [comment]
What to do if resolver returns multiple hostnames?

**Reply by fatalbanana** [reply]
> What to do if resolver returns multiple hostnames?

Pick the first one, Rspamd won't deal with multiple.

Hostnames passed to Rspamd are expected to be *forward-confirmed* by the MTA (so it should check if the label in question resolves back to the sending IP address).

**Reply by andy-igoshin** [reply]
> > What to do if resolver returns multiple hostnames?
> 
> Pick the first one, Rspamd won't deal with multiple.
> 
> Hostnames passed to Rspamd are expected to be _forward-confirmed_ by the MTA (so it should check if the label in question resolves back to the sending IP address).

I looked it up "src/plugins/lua/hfilter.lua" and I'm not sure about your thesis about the need for _forward-confirmed_.

In my opinion, the idea there is to check Hostname with “local weights = checks_hellohost_map:get_key(hostname)”.

Obviously Hostname not _forward-confirmed_ are good candidates for this check. Not to mention that having a valid reverse zone is not a requirement in current Internet.

**Comment by andy-igoshin** [comment]
If the forward-confirmed check fails, the Hostname header should not be set?

If the resolver returned multiple hostnames and one of them matched, should the _forward-confirmed_ check be considered successful?

**Reply by fatalbanana** [reply]
> If the forward-confirmed check fails, the Hostname header should not be set?

Yes.
 
> If the resolver returned multiple hostnames and one of them matched, should the _forward-confirmed_ check be considered successful?

Sure.
